clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -3104,15 +3107,37 @@ private int processKafkaDataMessage(
         // update checksum for this PUT message if needed.
         partitionConsumptionState.maybeUpdateExpectedChecksum(keyBytes, put);
 
-        // Do transorfmation recompute key, value and partition
+        // Check if put.getSchemaId is positive, if not default to 1
+        int putSchemaId = put.getSchemaId() > 0 ? put.getSchemaId() : 1;
+
+        // Do transformation recompute key, value and partition
         if (recordTransformer != null) {
           long recordTransformStartTime = System.currentTimeMillis();
+          ByteBuffer valueBytes = put.getPutValue();
+          Schema valueSchema = schemaRepository.getValueSchema(storeName, putSchemaId).getSchema();
```

Comment:
I got the unit test to reach here but schemaRepository.getValueSchema returns null. How would I set the schema from within the unit test?

Reply:
You can set up the schemaRepo in `StoreIngestionTaskFactory` via the builder pattern

Reply:
It looks like schemaRepo is being set in `getIngestionTaskFactoryBuilder`. How would I define the schema after schemaRepo has already been setup?

When I log `schemaRepository.getValueSchemas(storeName)` it returns an empty list.

Reply:
This has been resolved in my latest commit.

Reply:
The spotbug failed tho. Can you take a look?

Reply:
The CI has this error:
```
Execution failed for task ':clients:da-vinci-client:diffCoverage'.
> java.lang.Exception: Rule violated for bundle da-vinci-client: branches covered ratio is 0.3, but expected minimum is 0.4
```
But when I run it locally, it passes. Do you have an idea why?

<img width="356" alt="image" src="https://github.com/linkedin/venice/assets/35252220/b7cff3f4-227c-47f3-92d6-fb5e1d8cbdbe">

Reply:
It's very strange that JDK17 passes but JDK 11 and 8 failed. What's the JDK version on your local?

Reply:
I'm using JDK17 locally.

Reply:
This is very strange.. I pulled down your PR and ran it locally using JDK11. It passed w/o issue. Let us re-run the jobs.. In the meanwhile, please work on other open comments and maybe add a bit more tests to see if it helps??

Reply:
Okay, the retry worked